src/FSharpVSPowerTools.Logic/QuickInfoMargin.fs

```diff
@@ -37,51 +39,50 @@ type QuickInfoMargin (textDocument: ITextDocument,
     
     let buffer = view.TextBuffer
 
-    let updateQuickInfo (tooltip: string option, errors: ((FSharpErrorSeverity * string list) []) option) = lock updateLock <| fun () -> 
-        model.QuickInfo <-
-            errors
-            |> Option.map (fun errors ->
-                errors 
-                |> Array.map (fun (severity, errors) -> 
-                    let errors = List.map String.trim errors
-                    let title = 
-                        match errors with
-                        | [_] -> sprintf "%+A" severity
-                        | _ -> sprintf "%+As (%d)" severity errors.Length
-                    title + ": " + 
-                    (match errors with 
-                     | [e] -> e
-                     | _ -> 
-                        errors 
-                        |> List.mapi (fun i e -> sprintf "%d. %s" (i + 1) e) 
-                        |> List.toArray 
-                        |> String.concat " "))
-                |> String.concat " ")
-            |> Option.orElse (tooltip |> Option.bind (fun tooltip ->
-                tooltip 
-                |> String.getNonEmptyLines
-                |> Array.toList 
-                |> List.tryHead
-                |> Option.map (fun str ->
-                    if str.StartsWith("type ", StringComparison.Ordinal) then
-                        let index = str.LastIndexOf("=", StringComparison.Ordinal)
-                        if index > 0 then
-                            str.[0..index-1]
-                        else str
+    let updateQuickInfo (tooltip: string option, errors: ((FSharpErrorSeverity * string list) []) option) =
+        let updateFunc () =    
+            // helper function to lead a string builder across the collection of 
+            // errors accumulating lines annotated with their index number   
+            let errorString (errors:string list) (sb:StringBuilder) =
+                match errors with 
+                | [e] -> append e sb
+                | _ -> (sb, errors ) ||> List.foldi (fun sb i e -> 
+                    sb |> append (sprintf "%d. %s" (i + 1) e) |> append " ")
+                       
+            let currentInfo =
+                // if the tooltip contains errors show them
+                errors |> Option.map (fun errors ->
+                    (StringBuilder (), errors) ||> Array.fold (fun sb (severity, err) -> 
+                        let errorls = List.map String.trim err
+                        let title = 
+                            match errorls with
+                            | [_] -> sprintf "%s" <| string severity
+                            | _ -> sprintf "%s (%d)" (string severity) errorls.Length
+                        (sb |> append title |> append ": " |> errorString errorls |> append " ")) |> string)  
+                // show type info if there aren't any errors
+                |>  Option.orElse (tooltip |> Option.bind (fun tooltip ->
```

Comment:
We have `Option.orTry` function to do that. We haven't used this as much as we should.